### ISSUE TITLE
Evaluating a Trained FastRCNN in c#

### ISSUE THREAD

**Issue by Yandychang1** [Microsoft/CNTK#2507]

**Comment by ahoink** [comment]
Take a look at https://github.com/Microsoft/CNTK/tree/master/Examples/Image/Detection/FastRCNN/BrainScript/selectivesearch for generating ROIs

**Comment by Yandychang1** [comment]
Please correct me if I am wrong.... Each image will have its own Selective Search process done, and both the  output of the Selective Search and the image itself will be passed to the FastRCNN. Is this Correct?

**Comment by ahoink** [comment]
You are correct. Here is an example someone else made of evaluating a FastRCNN model trained using CNTK
https://github.com/CatalystCode/CNTK-FastRCNNDetector

**Comment by Yandychang1** [comment]
Thank you, I made it work. Now... I am trying to convert the output into bounding boxes and class probabilities. As I understand it: the output returns a Dictionary with two key,value pairs. the keys are:
1- Key: cls-pred, Value: NDArray of dimensions [3,200,1]
2-bbox-regr, Value: NDArray of dimensions [12,200,1]

I assume there are 2 classes( A and B) and a background class( 3 total)
When I get the DenseData from the output dictionary, I get two Lists, where position 0 of each List contains another List of Length 600 for cls-pred and 2400 for bbox-regr. 

1- I presume that each bbox-regr item contains 12 values needed to create the bounding box. my question is, How do I convert these values to coordinates to draw a rectangle on my picture.

2- I presume cls-pred has length 600. the reason is that there are 3 classes with positions 0 being background, 1 being class A, and 2 being Class B; and There is a softmax that determines the probabilities for each class.( would this be correct?)

again... Thank you for your help...

**Comment by ahoink** [comment]
your assumption (2) regarding the cls-pred is correct. the other output, bbox-regr, contains the regression coefficients per class for each ROI. You all three of these (cls-pred, bbox-regr, and ROIs), to filter out only the best bounding boxes (each should have 4 points to specify the location of each side of the box). 
The process of doing this is less specific to CNTK, but here is an example in C++ I used when I had the same problem you have (this is not my code): https://github.com/springkim/FasterRCNN_SpringEdition/blob/master/dev/CNTK-eval-CPU-only.cpp
Pay attention to the function ProcessResults()

**Comment by Yandychang1** [comment]
Thank you, I will look into it, appreciate the help
